### ISSUE TITLE
why is the "self.dim" have to be 7168

### ISSUE THREAD

**Issue by superamy666** [deepseek-ai/DeepSeek-V3#705]
`class Gate(nn.Module):
super().__init__()
self.dim = args.dim
        self.topk = args.n_activated_experts 
self.n_groups = args.n_expert_groups
        self.topk_groups = args.n_limited_groups
        self.score_func = args.score_func
        self.route_scale = args.route_scale
        self.weight = nn.Parameter(torch.empty(args.n_routed_experts, args.dim))
        self.bias = nn.Parameter(torch.empty(args.n_routed_experts)) if self.dim == 7168 else None`
why??

**Comment by AiMecreative** [comment]
I have the same question. note that $7168 = 7 \times 2^{10}$, and DeepSeek-v3 is trained on 2048 H800 GPUs, there may be some reasons
